### ISSUE TITLE
Have to relogin in everytime, what's going on with the cookies?

### ISSUE THREAD

**Issue by mayeaux** [sahat/hackathon-starter#733]
When I'm on my mobile device I have to re-login to my site everytime I access it. What's going on there? Thanks

**Comment by mayeaux** [comment]
https://stackoverflow.com/questions/18760461/nodejs-how-to-set-express-session-cookie-not-to-expire-anytime

Okay, just figured it out.

Maybe we don't want to do that here? Kind of a tough default setting..

**Comment by bassie1995** [comment]
For a production system, you should determine the cookie expire time based on, for example, sensitivity of the system: if the user left their PC unlocked and someone got into it, how big of an issue is that?

Apart from that, you'd probably want to offload sessions to another system (I use Redis) anyway, since the default memory store gets reset when the server exits and uses the server's RAM.

**Comment by niallobrien** [comment]
@bassie1995 I thought sessions were stored in Mongo...

**Comment by bassie1995** [comment]
Oh, whoops! I looked at that but somehow didn't register, serves me right for not taking my time.

It's weird then, yeah.

**Comment by mayeaux** [comment]
Default right now is to destroy the cookie at the end of the session. I'd like to recommend we let the cookie last for ~a week. People can reduce it if there's some kind of security needs but for a basic hackathon starter you shouldn't invalidate the cookie every browser session.

**Comment by niallobrien** [comment]
To be honest, I'm more concerned with the discarded cookies being left in the database, is there any way to automatically clean these up?

**Comment by mayeaux** [comment]
+1

**Comment by YasharF** [comment]
While looking into this issue, I think I found a different bug at https://github.com/sahat/hackathon-starter/blob/master/app.js#L76 but I need someone else to verify it as well.  There is `clear_interval: 3600` which if I understood it correctly was meant for the session data to expire and remove from the DB in 3600 seconds (1 hour).  I could not find documentation on  `clear_interval` but I did found: 

- https://github.com/jdesboeufs/connect-mongo/issues/46 
- https://github.com/jdesboeufs/connect-mongo/pull/50
- https://github.com/jdesboeufs/connect-mongo/commit/9445612b1bb674cff021a8d1fccc985b153d3e6f 

Pointing to `clear_interval` being deprecated a few years ago.  Can someone else confirm?  If so, it looks like we need to update those few lines.

Can someone submit a PR?
